api/review_checklist.md

```diff
@@ -0,0 +1,137 @@
+# API Review Checklist
```

Comment:
What about adding a checkbox to https://github.com/envoyproxy/envoy/blob/master/PULL_REQUEST_TEMPLATE.md indicating that (for API changes) that they've understood this policy?

Reply:
Done.